src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -525,9 +527,11 @@ private void LoadExtensions(Assembly assembly, string locationHint)
             if (!string.IsNullOrEmpty(instrumentationKey))
             {
                 metrics.LogEvent(MetricEventNames.ApplicationInsightsEnabled);
-                ScriptTelemetryClientFactory factory = new ScriptTelemetryClientFactory();
-                scriptConfig.HostConfig.LoggerFactory
-                    .AddApplicationInsights(instrumentationKey, scriptConfig.LogFilter.Filter, factory, scriptConfig.ApplicationInsightsSamplingSettings);
+
+                ITelemetryClientFactory factory = scriptConfig.HostConfig.GetService<ITelemetryClientFactory>() ??
+                    new ScriptTelemetryClientFactory(instrumentationKey, scriptConfig.ApplicationInsightsSamplingSettings);
```

Comment:
My e2e tests (coming next) will override this factory. Adding this in preparation.